IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m

```diff
@@ -444,16 +438,60 @@ - (BOOL)saveAppMetadata:(nonnull MSIDAppMetadataCacheItem *)metadata
                                   error:error];
 }
 
-- (nullable MSIDAppMetadataCacheItem *)getAppMetadata:(nonnull MSIDAppMetadataCacheKey *)key
-                                              context:(nullable id<MSIDRequestContext>)context
-                                                error:(NSError * _Nullable * _Nullable)error
+- (nullable MSIDAppMetadataCacheItem *)getAppMetadataWithQuery:(nonnull MSIDAppMetadataCacheQuery *)query
```

Comment:
Since this API returns a single item, I think it has to be more specific about what it returns. I would say it should reject any query that's no exact match and only lookup one particular item (or even better accept MSIDAppMetadataCacheKey and not query). Otherwise we return first item from an indeterministic query. And the method that returns multiple items can do additional matching when necessary then and return multiple items based on the query.

Reply:
It cannot take a key because we need to have environment aliases when we try to look for app metadata and as discussed we cannot add this property to the key. Lets discuss this offline.